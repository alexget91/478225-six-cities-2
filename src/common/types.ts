import {PlaceType} from "./constants";

type Host = {
  id: number,
  name: string,
  avatar_url?: string,
  is_pro?: boolean,
}

type User = {
  email: string,
} & Host;

type Location = {
  latitude: number,
  longitude: number,
  zoom?: number,
}

type City = {
  name: string,
  location: Location,
};

type PlaceCard = {
  id: number,
  city?: City,
  preview_image?: string,
  images?: Array<string>,
  title: string,
  is_premium?: boolean,
  is_favorite?: boolean,
  rating?: number,
  type: PlaceType,
  bedrooms?: number,
  max_adults?: number,
  price: number,
  goods?: Array<string>,
  host?: Host,
  description?: string,
  location?: Location,
};

type PlaceList = Array<PlaceCard>;

type ReviewItem = {
  id: number,
  user: Host,
  rating: number,
  comment: string,
  date: string,
};

type ReviewsList = Array<ReviewItem>;

type PinCoordinates = [number, number];

type Pin = {
  coords: PinCoordinates,
  isActive?: boolean,
}

type AnyProps = {
  [propName: string]: any
};

export {User, City, PlaceCard, PlaceList, ReviewItem, ReviewsList, PinCoordinates, Pin, AnyProps};
