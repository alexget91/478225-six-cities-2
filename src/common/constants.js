const pageTypes = {
  FAVORITES: `FAVORITES`,
  FAVORITES_EMPTY: `FAVORITES_EMPTY`,
  LOGIN: `LOGIN`,
  MAIN: `MAIN`,
  MAIN_EMPTY: `MAIN_EMPTY`,
  OFFER: `OFFER`,
};

const sortingOptions = {
  popular: `popular`,
  toHigh: `to-high`,
  toLow: `to-low`,
  topRated: `top-rated`,
};

export {pageTypes, sortingOptions};
