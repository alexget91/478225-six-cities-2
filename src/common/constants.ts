enum PageType {
  FAVORITES = `FAVORITES`,
  FAVORITES_EMPTY = `FAVORITES_EMPTY`,
  LOGIN = `LOGIN`,
  MAIN = `MAIN`,
  MAIN_EMPTY = `MAIN_EMPTY`,
  OFFER = `OFFER`,
}

enum PlaceType {
  APARTMENT = `apartment`,
  ROOM = `room`,
  HOUSE = `house`,
  HOTEL = `hotel`,
}

enum PlacesListView {
  OFFER = `OFFER`,
  LIST = `LIST`,
  FAVORITES = `FAVORITES`,
}

enum SortingOption {
  POPULAR = `popular`,
  TO_HIGH = `to-high`,
  TO_LOW = `to-low`,
  TOP_RATED = `top-rated`,
}

enum FormSendingStatus {
  READY = `READY`,
  SUCCESS = `SUCCESS`,
  SENDING = `SENDING`,
}

export {PageType, PlaceType, PlacesListView, SortingOption, FormSendingStatus};
