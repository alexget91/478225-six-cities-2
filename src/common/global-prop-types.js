import PropTypes from "prop-types";

const placeCard = {
  id: PropTypes.number.isRequired,
  previewImage: PropTypes.string,
  images: PropTypes.arrayOf(PropTypes.string),
  title: PropTypes.string.isRequired,
  isPremium: PropTypes.bool,
  isFavorite: PropTypes.bool,
  rating: PropTypes.number,
  type: PropTypes.oneOf([`apartment`, `room`, `house`, `hotel`]).isRequired,
  bedrooms: PropTypes.number,
  maxAdults: PropTypes.number,
  priceByNight: PropTypes.number.isRequired,
  goods: PropTypes.arrayOf(PropTypes.string),
  host: PropTypes.exact({
    isPro: PropTypes.bool,
    name: PropTypes.string,
    avatarUrl: PropTypes.string
  }),
  description: PropTypes.string,
};

const placeList = PropTypes.arrayOf(PropTypes.exact(placeCard));

export {placeCard, placeList};
