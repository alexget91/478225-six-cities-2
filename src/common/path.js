const Path = {
  INDEX: `/`,
  OFFER: `/offer`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
};

export default Path;
