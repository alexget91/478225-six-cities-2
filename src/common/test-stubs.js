const createMapBlock = () => {
  const div = global.document.createElement(`div`);
  div.setAttribute(`id`, `map`);
  global.document.body.appendChild(div);
};

const getMockOffer = (id, cityName) => {
  return {
    id,
    city: {
      name: cityName,
      location: {
        latitude: 0,
        longitude: 0,
      }
    },
    priceByNight: 0,
    title: `Title${id}`,
    type: `apartment`,
    location: {
      latitude: 0,
      longitude: 0,
    }
  };
};

export {createMapBlock, getMockOffer};
