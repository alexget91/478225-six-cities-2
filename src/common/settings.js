import {sortingOptions} from "./constants";

const Settings = {
  API_TIMEOUT: 5000,
  DEFAULT_SORT: sortingOptions.popular,
  COMMENT_LENGTH_MIN: 50,
  COMMENT_LENGTH_MAX: 300,
};

export default Settings;
