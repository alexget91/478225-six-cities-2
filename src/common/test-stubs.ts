import {PlaceType} from "./constants";
import {PlaceCard} from "./types";

const createMapBlock = (): void => {
  const div: HTMLDivElement = (global as any).document.createElement(`div`);
  div.setAttribute(`id`, `map`);
  (global as any).document.body.appendChild(div);
};

const getMockOffer = (id: number, cityName: string, isFavorite?: boolean): PlaceCard => ({
  id,
  city: {
    name: cityName,
    location: {
      latitude: 0,
      longitude: 0,
    },
  },
  is_favorite: isFavorite || false,
  price: 0,
  title: `Title${id}`,
  type: PlaceType.APARTMENT,
  location: {
    latitude: 0,
    longitude: 0,
  },
});

export {
  createMapBlock,
  getMockOffer,
};
