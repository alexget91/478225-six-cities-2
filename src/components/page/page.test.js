import React from "react";
import renderer from "react-test-renderer";
import Page from "./page";
import {pageTypes} from "../../common/constants";

describe(`Page component correctly renders after relaunch`, () => {
  it(`Without data`, () => {
    const tree = renderer
      .create(<Page/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Without type`, () => {
    const tree = renderer
      .create(<Page
        header={<div className="header"></div>}
        content={<div className="content"></div>}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With type FAVORITES`, () => {
    const tree = renderer
      .create(<Page
        header={<div className="header"></div>}
        content={<div className="content"></div>}
        type={pageTypes.FAVORITES}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With type FAVORITES_EMPTY`, () => {
    const tree = renderer
      .create(<Page
        header={<div className="header"></div>}
        content={<div className="content"></div>}
        type={pageTypes.FAVORITES_EMPTY}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With type LOGIN`, () => {
    const tree = renderer
      .create(<Page
        header={<div className="header"></div>}
        content={<div className="content"></div>}
        type={pageTypes.LOGIN}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With type MAIN`, () => {
    const tree = renderer
      .create(<Page
        header={<div className="header"></div>}
        content={<div className="content"></div>}
        type={pageTypes.MAIN}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With type MAIN_EMPTY`, () => {
    const tree = renderer
      .create(<Page
        header={<div className="header"></div>}
        content={<div className="content"></div>}
        type={pageTypes.MAIN_EMPTY}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With type OFFER`, () => {
    const tree = renderer
      .create(<Page
        header={<div className="header"></div>}
        content={<div className="content"></div>}
        type={pageTypes.OFFER}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
