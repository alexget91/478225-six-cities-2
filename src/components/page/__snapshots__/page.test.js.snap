// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component correctly renders after relaunch With error message 1`] = `
<div
  className="page"
>
  <div
    className="header"
  />
  <div
    style={
      Object {
        "backgroundColor": "red",
        "color": "black",
        "fontSize": "25px",
        "left": 0,
        "margin": "0 auto",
        "padding": "5px",
        "position": "fixed",
        "right": 0,
        "textAlign": "center",
        "top": 0,
        "zIndex": 100,
      }
    }
  >
    error message
    <span
      className="js-close"
      style={
        Object {
          "cursor": "pointer",
          "float": "right",
        }
      }
    >
      [close]
    </span>
  </div>
  <main
    className="page__main"
  >
    <div
      className="content"
    />
  </main>
</div>
`;

exports[`Page component correctly renders after relaunch With type FAVORITES 1`] = `
<div
  className="page"
>
  <div
    className="header"
  />
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="content"
    />
  </main>
</div>
`;

exports[`Page component correctly renders after relaunch With type FAVORITES_EMPTY 1`] = `
<div
  className="page page--favorites-empty"
>
  <div
    className="header"
  />
  <main
    className="page__main page__main--favorites page__main--favorites-empty"
  >
    <div
      className="content"
    />
  </main>
</div>
`;

exports[`Page component correctly renders after relaunch With type LOGIN 1`] = `
<div
  className="page page--gray page--login"
>
  <div
    className="header"
  />
  <main
    className="page__main page__main--login"
  >
    <div
      className="content"
    />
  </main>
</div>
`;

exports[`Page component correctly renders after relaunch With type MAIN 1`] = `
<div
  className="page page--gray page--main"
>
  <div
    className="header"
  />
  <main
    className="page__main page__main--index"
  >
    <div
      className="content"
    />
  </main>
</div>
`;

exports[`Page component correctly renders after relaunch With type MAIN_EMPTY 1`] = `
<div
  className="page page--gray page--main"
>
  <div
    className="header"
  />
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <div
      className="content"
    />
  </main>
</div>
`;

exports[`Page component correctly renders after relaunch With type OFFER 1`] = `
<div
  className="page"
>
  <div
    className="header"
  />
  <main
    className="page__main page__main--property"
  >
    <div
      className="content"
    />
  </main>
</div>
`;

exports[`Page component correctly renders after relaunch Without data 1`] = `
<div
  className="page"
>
  <main
    className="page__main"
  />
</div>
`;

exports[`Page component correctly renders after relaunch Without type 1`] = `
<div
  className="page"
>
  <div
    className="header"
  />
  <main
    className="page__main"
  >
    <div
      className="content"
    />
  </main>
</div>
`;
